rusk-wallet/src/wallet/file_service.rs

```diff
@@ -0,0 +1,341 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::path::PathBuf;
+
+use wallet_core::Seed;
+
+use crate::crypto::decrypt;
+use crate::dat::DatFileVersion;
+use crate::Error;
+
+/// Provides access to a secure wallet file
+pub trait SecureWalletFile: Debug + Send + Sync + Clone {
+    /// The type of the path buffer wrapper
+    type PathBufWrapper: WalletFilePath + Hash + Eq + PartialEq + Debug + Clone;
+
+    // Methods to implement ===================================================
+
+    /// Returns the path
+    fn path(&self) -> &Self::PathBufWrapper;
+    /// Return the mutable path
+    fn path_mut(&mut self) -> &mut Self::PathBufWrapper;
+    /// Returns the hashed password
+    fn pwd(&self) -> &[u8];
+    /// Returns the file version
+    fn version(&self) -> DatFileVersion;
+
+    // Automatically implemented methods =======================================
+
+    /// Returns the path of the wallet file
+    fn wallet_path(&self) -> &PathBuf {
+        self.path().wallet_path()
+    }
+
+    /// Returns the directory of the profile
+    fn profile_dir(&self) -> &PathBuf {
+        self.path().profile_dir()
+    }
+
+    /// Returns the network name for different cache locations
+    /// e.g, devnet, testnet, etc.
+    fn network(&self) -> Option<&String> {
+        self.path().network()
+    }
+
+    /// Sets the network name for different cache locations.
+    /// e.g, devnet, testnet, etc.
+    fn set_network_name(&mut self, network: Option<String>) {
+        self.path_mut().set_network_name(network);
+    }
+
+    /// Returns the filename of this file
+    fn name(&self) -> Option<String> {
+        self.path().name()
+    }
+
+    /// Returns dir for cache based on network specified
+    fn cache_dir(&self) -> PathBuf {
+        self.path().cache_dir()
+    }
+
+    /// Checks if the file version is older than the latest Rust Binary file
+    /// format
+    fn is_old(&self) -> bool {
+        let version = self.version();
+        matches!(
+            version,
+            DatFileVersion::Legacy | DatFileVersion::OldWalletCli(_)
+        )
+    }
+
+    /// Get the seed and address from the file
+    fn get_seed_and_address(&self) -> Result<(Seed, u8), Error> {
```

Comment:
This logic should be inside the `file.rs` where the rest of `DatFileVersion` structs and magic numbers are defined